R/utils.R

```diff
@@ -21,6 +21,15 @@ nan_is_na = function(x) {
   stopf("Argument 'nan' must be NA or NaN")
 }
 
+# In R 3.3.0, rep_len() strips attributes --> breaks data.table()'s internal recycle() helper.
+if (inherits(rep_len(Sys.Date(), 1L), "Date")) {
```

Comment:
I just checked and this seems still a problem at `3.6.0` so we might need to hold on to this for quite some time, but it would be best to check at which version this is fixed in `R` and add a comment

Reply:
Thanks for checking. I suspected that might be the case -- I checked the R NEWS and NEWS.3 files and didn't see anything relevant here (searching 'attrib', 'rep_len'), and glanced at the `rep_len()` blame to see if anything about attributes popped out, no luck.

Reply:
Actually looking again, I think it's this:

https://github.com/r-devel/r-svn/commit/70e8a3fd55b8f4e5b87d5fbbe7256e6f59154961

Then `rep_len()` on `Date` dispatches to `rep.Date()`.

That puts us in R 4.0.0:

https://github.com/r-devel/r-svn/blob/874fa1008e6cb090c7c898c422d703ca0026d600/doc/NEWS.Rd#L4885-L4887